BUILD_macos.md

```diff
@@ -1,122 +1,182 @@
-  #
-  # into ~/.zprofile and run:
-  #
-  # . ~/.zprofile
-  #
-  # or open a new shell
-  ```
-
-  * Install the required version
-
-  ```shell
-  rustup toolchain install <version from build.assets/versions.mk>
-  cd <teleport.git>
-  rustup override set <version from build.assets/versions.mk>
-  rustc --version
-  # rustc <version from build.assets/versions.mk>
-  ```
-
-* To install `libfido2` (pulls `openssl 3` as dependency)
-
-  ```shell
-  brew install libfido2
-  ```
-
-* To install `pkg-config`
-
-  ```shell
-  brew install pkg-config
-  ```
-
-* To install tools for building the UI:
-  * `brew install node corepack`
-  * `corepack enable pnpm`
-  * The `Rust` and `Cargo` version in [build.assets/Makefile](https://github.com/gravitational/teleport/blob/master/build.assets/versions.mk#L11) (search for `RUST_VERSION`) are required.
-  * The [`wasm-pack`](https://github.com/rustwasm/wasm-pack) version in [build.assets/Makefile](https://github.com/gravitational/teleport/blob/master/build.assets/versions.mk#L12) (search for `WASM_PACK_VERSION`) is required:
-    `curl https://rustwasm.github.io/wasm-pack/installer/init.sh -sSf | sh`
-
-##### Local Tests Dependencies
-
-To run a full test suite locally, you will need
-
-* `helm` and `helm-unittest` plugin
-
-  ```shell
-  brew install helm
-  helm plugin install https://github.com/quintush/helm-unittest
-  ```
-
-* `bats-core` version from [build.assets/Dockerfile](https://github.com/gravitational/teleport/blob/master/build.assets/Dockerfile#L183) (search for `bats-core`)
-
-  ```shell
-  curl -L https://github.com/bats-core/bats-core/archive/v1.2.1.tar.gz -o ~/Downloads/bats.tar.gz
-  cd ~/Downloads
-  tar xzvf bats.tar.gz
-  sudo mkdir /usr/local/libexec
-  sudo chown $USER /usr/local/libexec
-  cd bats-core-1.2.1
-  sudo ./install.sh /usr/local
-  cd ../
-  rm -rf bats-core-1.2.1 bats.tar.gz
-  ```
-
-* `protoc` binary, typically found in `protobuf` package
-
-  ```shell
-  brew install protobuf
-  ```
-
-* increased `ulimit -n`
-
-  ```shell
-  ulimit -n 2560 # 10x default
-  ```
+# MacOS Environment Setup
+
+To set up your MacOS environment, follow these steps using Homebrew as the main
+package manager. Aim to install versions specified in
+[`build.assets/versions.mk`](/build.assets/versions.mk); for others, use the
+latest Homebrew version.
+
+The instructions below are provided on a best-effort basis. PRs with corrections
+and updates are welcome!
+
+1. **Install [Homebrew](https://brew.sh/)**
+
+1. **Install Go**
+    1. Find the required Go version in
+      [`build.assets/versions.mk`](/build.assets/versions.mk)
+      (`GOLANG_VERSION`).
+
+    1. Install Go (Homebrew only supports MAJOR.MINOR versions):
+
+        ```shell
+        # Replace <version> with the MAJOR.MINOR value of GOLANG_VERSION from build.assets/versions.mk (e.g., 1.24)
+        brew install go@<version>
+        ```
+
+    1. Verify the installed version:
+
+        ```shell
+        go version
+        ```
```

Comment:
All recent versions of the Go toolchain should be able to find the right version from go.mod and install and run that version automatically, so we can probably simplify this to just `brew install go`